### ISSUE TITLE
[enhancement] Reuse the longest safe common token prefix across divergent chat requests

### ISSUE THREAD

**Issue by rjckkkkk** [Approaching-AI/AIMA-AMD395-Qwen36-35B-Linux-Engine#12]
## Problem

The native prefix cache only reuses a cached entry when the **entire cached request token sequence** is an exact prefix of the new request. This makes ordinary chat-prefix reuse miss even when two requests share a substantial system/user prefix.

Observed on `v1.5.1-native-vl.6`:

1. Send system `You are a helpful assistant.` + user `你好`: cache miss.
2. Repeat the same request: exact cache hit.
3. Send the same system message + user `你好，你是谁`: cache miss with `matched_tokens: 0`.

`chat-template-probe` shows that the prompts share 15 leading tokens, then diverge because the shorter request inserts the user-message/chat-template terminator where the longer request continues:

```text
"你好"         ... 109266, 248046, 198, 248045, 74455, ...
"你好，你是谁" ... 109266,   3709, 144810, 248046, 198, ...
```

The current behavior follows `native_prefix_cache_matched_tokens()`: it compares all cached tokens against the beginning of the request and returns zero for any shorter common prefix. `NativeExactPrefixCache::capture()` also captures only the completed prompt state.

This is consistent with the current exact request-prefix product contract, but it regresses the practical prefix-cache behavior expected by users migrating from a block/radix cache.

## Requested capability

Please implement longest safe common-token-prefix reuse for ordinary Chat Completions requests, including prefixes that end before the cached request's terminal token.

For the hybrid Qwen3.6 model, a reusable boundary must preserve all state needed to resume exactly:

- full-attention KV up to the matched boundary;
- linear-attention recurrent and convolution state at that boundary;
- the correct hidden/input boundary for suffix prefill;
- multimodal namespace and media-identity isolation;
- LRU/capacity behavior within the existing 96 GiB GTT contract.

The implementation can use block checkpoints, message-boundary checkpoints, or another design, provided it never restores state past the exact common-token boundary.

## Acceptance criteria

- The reproduction above reports a non-zero prefix hit for request 3 and prefills only the unmatched suffix.
- Exact-repeat and existing whole-request append hits remain correct.
- Diverging text is never treated as matched; reported `matched_tokens` reflects the state actually restored.
- Text, multi-turn chat, and multimodal cache namespaces remain isolated correctly.
- Output token identity/correctness gates pass against cold execution for exact and partial hits.
- Add CPU tests for longest-prefix selection and AMD395 resident A/B tests covering exact, append, divergent chat, eviction, and media identity.
- Report TTFT, restore bytes/time, suffix work, decode retention, and peak memory for the promoted configuration.

@skyguan92 please own the engine design and implementation because the current whole-prompt state capture and lookup were introduced in the prefix-cache path you authored.

**Comment by skyguan92** [comment]
Implemented in [PR #4](https://github.com/skyguan92/AIMA-AMD395-Qwen36-35B-Linux-Engine/pull/4) and published as [v1.5.1-native-vl.7](https://github.com/skyguan92/AIMA-AMD395-Qwen36-35B-Linux-Engine/releases/tag/v1.5.1-native-vl.7). Both repository main branches and the immutable release tag are synchronized.

The reported `你好` → `你好，你是谁` case now restores 15 common prompt tokens and prefills only the remaining 11. Saved checkpoints restore KV, recurrent, convolution and hidden state together; media identity and LRU/capacity isolation remain enforced.

Qualification with the exact engine and pinned portable runtime passed:

- Both cache profiles: 52 generation pairs and 68 full-vocabulary logits comparisons, independently recomputed from the published raw evidence; maximum KLD 0.0035071 and all compared top-1 predictions match.
- All 19 text-matrix cells, 11 chat-protocol checks, 12 HTTP checks, isolated archive execution and exact v1.5.1 rollback.
- A full one-hour, 360-request mixed-workload soak, with one model load, zero post-warm RSS growth and clean shutdown.
- Repository, security, evidence and final release gates, plus CI for the merged commit and release tag.

All four public release assets were downloaded anonymously and checksum-verified. The downloaded engine's embedded source identity and all 209 pinned runtime entries were checked on Linux; the downloaded evidence archive passed standalone verification.

This implements the longest **saved eligible** common-prefix checkpoint, not an arbitrary-token radix cache or universal BF16 bitwise equivalence. Longer checkpoints use 32-token alignment, and unsafe short cross-block continuations fall back to cold execution. See the [release provenance and documented limits](https://github.com/skyguan92/AIMA-AMD395-Qwen36-35B-Linux-Engine/blob/main/docs/RELEASE.md).
